include/dl/dl_typelib.h

```diff
@@ -19,19 +19,33 @@ extern "C" {
 
 #include <dl/dl.h>
 
+/*
+	Function pointer type: dl_include_handler
+	    Is responsible for locating and loading a type-library into dl_ctx.
+
+	Parameters:
+	    handler_ctx     - context passed in to dl_context_load_txt_type_library_with_includes()
+	    dl_ctx          - dl-context to load typelib into.
+	    file_to_include - the path to the file to include
+ */
+typedef dl_error_t ( *dl_include_handler )( void* handler_ctx, dl_ctx_t dl_ctx, const char* file_to_include );
```

Comment:
Pass a struct with ctx and include-path/id and user data instead, this to make api easier to extend.

Reply:
I know where you are coming from, but in this case I disagree for one important reason, this is a function which DL calls and the user of dl implements.

I want a compile error when the signature changes. Otherwise I might miss a change and thus not implement support for the functionality the new argument. If this had been a function which DL implemented and only the called would have to pass in a struct which can be default constructed then I would agree with you.